### PR TITLE
[9.0][ADD] Migrated business_requirement_deliverable_cost module from v8 to v9

### PR DESCRIPTION
**Pull request by sudhir-serpentcs**
## WIP until #25  and #26  is merged

Migrated business_requirement_deliverable_cost module from v8 to v9.
WIP: Clean up of unit test cases.
@dreispt @pedrobaeza @jbeficent @moylop260 Please review.

### CODE REVIEW

File: business_requirement_deliverable_cost/README.rst
Comment (elicoidal) [inline business_requirement_deliverable_cost/README.rst]: 9.0

**Review by elicoidal — CHANGES_REQUESTED** [review]
:+1: 
I will update the README in separate file

File: business_requirement_deliverable_cost/__openerp__.py
Comment (elicoidal) [inline business_requirement_deliverable_cost/__openerp__.py:16]: add icon image

**Review by elicoidal — APPROVED** [review]
:+1: 
Small details
@pedrobaeza @moylop260 @mmalorni

**Comment by sudhir-serpentcs** [comment]
Rebased the PR.
@dreispt @pedrobaeza @jbeficent @moylop260 @elicoidal

File: business_requirement_deliverable_cost/models/business.py
Comment (astirpe) [inline business_requirement_deliverable_cost/models/business.py]: Please apply the same fix that was made for V8 in https://github.com/OCA/business-requirement/pull/66/files

**Review by astirpe — CHANGES_REQUESTED** [review]

File: business_requirement_deliverable_cost/models/business.py
Comment (elicoidal) [reply business_requirement_deliverable_cost/models/business.py]: @sudhir-serpentcs Can you attend @astirpe recommendation so that we can merge cleanly?

Comment (sudhir-serpentcs) [reply business_requirement_deliverable_cost/models/business.py]: @astirpe @elicoidal Done

**Comment by elicoidal** [comment]
@astirpe @sudhir-serpentcs can we merge this one?

**Comment by sudhir-serpentcs** [comment]
I think yes. We can merge it.
@elicoidal

**Comment by astirpe** [comment]
With commit 6ebfc4b2c83be6272f611720cf3f18af1daf46f8 the method `def _get_partner()` was replaced with a related field `partner_id`. Is it possible to apply the same change before to merge? Otherwise we merge now and then open a new PR for that. I think doing now is the quickest way. What do you think?

**Comment by elicoidal** [comment]
agree with @astirpe indeed.
@sudhir-serpentcs could you do that and then we move forward?

**Comment by sudhir-serpentcs** [comment]
`With commit 6ebfc4b the method def _get_partner() was replaced with a related field partner_id. Is it possible to apply the same change before to merge? Otherwise we merge now and then open a new PR for that. I think doing now is the quickest way. What do you think?`

@elicoidal @astirpe Replaced the _get_partner() method by related partner_id field.

File: business_requirement_deliverable_cost/models/business.py
Comment (elicoidal) [inline business_requirement_deliverable_cost/models/business.py]: please remove dead codes

File: business_requirement_deliverable_cost/security/business_requirement_deliverable_security.xml
Comment (elicoidal) [inline business_requirement_deliverable_cost/security/business_requirement_deliverable_security.xml]: Can you improve the format of the lines for readibility?

**Review by elicoidal — CHANGES_REQUESTED** [review]
small details then we can move forward

**Review by astirpe — APPROVED** [review]
For me is OK to merge, after the removal of dead code ...

**Comment by sudhir-serpentcs** [comment]
@elicoidal @astirpe Removed dead code and improved security xml.

**Review by elicoidal — APPROVED** [review]

**Pull request merged**
